src/index.rs

```diff
@@ -26,12 +26,50 @@ impl From<SortOrder> for Bson {
     }
 }
 
+#[derive(Clone, Debug)]
+enum IndexKey {
+    SortIndex(SortIndexKey),
+    TextIndex(TextIndexKey),
+}
+
+impl IndexKey {
+    pub fn get_key_name(&self) -> String {
+        match self {
+            IndexKey::SortIndex(s) => match s.direction {
+                SortOrder::Ascending => format!("{}_1", s.name),
+                SortOrder::Descending => format!("{}_-1", s.name),
+            },
+
+            IndexKey::TextIndex(t) => format!("{}_text", t.name),
+        }
+    }
+
+    pub fn get_name(&self) -> String {
+        match self {
+            IndexKey::SortIndex(s) => s.name.to_string(),
+            IndexKey::TextIndex(t) => t.name.to_string(),
+        }
+    }
+
+    pub fn get_value(&self) -> Bson {
+        match self {
+            IndexKey::SortIndex(s) => s.direction.into(),
+            IndexKey::TextIndex(_) => "text".into(),
+        }
+    }
```

Comment:
Should these be private functions?

Reply:
Good point, I thought Rust would not like it if I tried to use them outside of the impl but it seems to work, will commit.